amx/server/syscalls.lua

```diff
@@ -2763,15 +2763,37 @@ function SetPlayerChatBubble(amx, player, text, color, dist, exptime)
 end
 
 
--- In MTA we can't set the type of the request. 
--- When we add 'data' it becomes POST request.
--- TODO: arguments "index" and "type" here only for compatibility.
+-- Now we have all of RESTful types of requests. Our function is better!
+-- The SAMP documentation said about 'url' - "The URL you want to request. (Without 'http://')"
+-- But I didn't set it up. Because WE SUPPORT HTTPS!
+-- TODO: An "index" argument only for compatibility.
 function HTTP(amx, index, type, url, data, callback)
-	if type == 3 then 
-		notImplemented('HTTP', 'A HEAD type have not support right now')
-	end
-	local request = fetchRemote ("http://".. url, 
-	function(responseData, error)
+
+	local typesToText = {
+		'GET',
+		'POST',
+		'HEAD',
+		'PUT',
+		'PATCH',
+		'DELETE',
+		'COPY',
+		'OPTIONS',
+		'LINK',
+		'UNLINK',
+		'PURGE',
+		'LOCK',
+		'UNLOCK',
+		'PROPFIND',
+		'VIEW'

```

Comment:
I like the idea of supporting extra kinds, but what if SAMP gets an update that supports extra methods?

Reply:
> 
> 
> I like the idea of supporting extra kinds, but what if SAMP gets an update that supports extra methods?

The last major update of samp was 5 years ago. If it will, I'm personally add it.

Reply:
Okay I guess an update is unlikely :)

But what I meant is that it may not be added in this order, and that it would cause a nasty problem if SAMP decides that `4` is "PATCH" instead of "PUT".

What if you used negative numbers for the extras? That would work around the problem

Reply:
> 
> 
> Okay I guess an update is unlikely :)
> 
> But what I meant is that it may not be added in this order, and that it would cause a nasty problem if SAMP decides that `4` is "PATCH" instead of "PUT".
> 
> What if you used negative numbers for the extras? That would work around the problem

Thanks for the advice. I will use negative numbers.